src/Core/AdminConsole/Services/Implementations/RabbitMqEventListenerBase.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Text;
+using System.Text.Json;
+using Bit.Core.Models.Data;
+using Bit.Core.Settings;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace Bit.Core.Services;
+
+public abstract class RabbitMqEventListenerBase : BackgroundService
+{
+    private IChannel _channel;
+    private IConnection _connection;
+    private readonly string _exchangeName;
+    private readonly ConnectionFactory _factory;
+    private readonly ILogger<RabbitMqEventListenerBase> _logger;
+
+    protected abstract string QueueName { get; }
+
+    protected RabbitMqEventListenerBase(
+        ILogger<RabbitMqEventListenerBase> logger,
+        GlobalSettings globalSettings)
+    {
+        _factory = new ConnectionFactory
+        {
+            HostName = globalSettings.RabbitMq.HostName,
+            UserName = globalSettings.RabbitMq.Username,
+            Password = globalSettings.RabbitMq.Password
+        };
+        _exchangeName = globalSettings.RabbitMq.ExchangeName;
+        _logger = logger;
+    }
+
+    public override async Task StartAsync(CancellationToken cancellationToken)
+    {
+        _connection = await _factory.CreateConnectionAsync(cancellationToken);
+        _channel = await _connection.CreateChannelAsync(cancellationToken: cancellationToken);
+
+        await _channel.ExchangeDeclareAsync(exchange: _exchangeName, type: ExchangeType.Fanout);
+        await _channel.QueueDeclareAsync(queue: QueueName,
+                                         durable: false,
+                                         exclusive: false,
+                                         autoDelete: false,
+                                         arguments: null,
+                                         cancellationToken: cancellationToken);
+        await _channel.QueueBindAsync(queue: QueueName, exchange: _exchangeName, routingKey: "");
+        await base.StartAsync(cancellationToken);
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        var consumer = new AsyncEventingBasicConsumer(_channel);
+        consumer.ReceivedAsync += async (model, eventArgs) =>
+        {
+            var body = eventArgs.Body.ToArray();
+            var message = Encoding.UTF8.GetString(body);
+
+            try
+            {
+                var eventMessage = JsonSerializer.Deserialize<EventMessage>(message);
+                await HandleMessageAsync(eventMessage);
+            }
```

Comment:
JsonSerializer is going to convert the string back into bytes anyways, no need to allocate an intermediary string just for it to be thrown away.

```suggestion
            try
            {
                var eventMessage = JsonSerializer.Deserialize<EventMessage>(eventArgs.Body.Span);
                await HandleMessageAsync(eventMessage);
            }
```